### DISCUSSION TITLE
Help with audio library

### DISCUSSION THREAD

**Discussion by Skyfrei** [ppy/osu#30491] [Q&A] [answered]
Hi,

I am thinking of a custom ruleset but I would need to go and control the audio devices of the system. Could you please tell me which audio library osu is using since I am trying to have it be compatible for both mac and windows?

**Comment by Skyfrei** [comment]
I did take a look at Bass(since ppy wrote this in a reddit post) but this is only free for non commercial products, and I am unsure if osu counts as non commercial

**Comment by cl8n** [comment]
not an answer but I'm curious to know why you need to do this in a ruleset, this feels like stepping way outside of the expected responsibilities of a ruleset to me.

**Reply by Skyfrei** [reply]
Well I would need access to input drivers for audio so I can make a ruleset getting input from an instrument

**Comment by peppy** [comment, marked as answer]
Sounds like you may need to do some more reading yourself.
